include/nbl/video/utilities/SDefaultPhysicalDeviceFilter.h

```diff
@@ -38,8 +38,43 @@ namespace nbl::video
 
         bool meetsRequirements(const IPhysicalDevice * const physicalDevice) const
         {
-            if(physicalDevice->getProperties().apiVersion < minApiVersion)
+            const auto& properties = physicalDevice->getProperties();
+            const auto& physDevLimits = physicalDevice->getProperties().limits;
+            const auto& physDevFeatures = physicalDevice->getFeatures();
+            const auto& memoryProps = physicalDevice->getMemoryProperties();
+            const auto& queueProps = physicalDevice->getQueueFamilyProperties();
+
+            if (properties.apiVersion < minApiVersion)
+                return false;
+            if (!deviceTypeMask.hasFlags(properties.deviceType))
+                return false;
+            if (!driverIDMask.hasFlags(properties.driverID))
```

Comment:
`hasFlags` is an equality, check right? not a subset?

Reply:
I'm pretty sure everything is correct here, `properties.deviceType` must be subset of `driverIDMask`
```c
inline bool hasFlags(bitflag<ENUM_TYPE> val) const { return (value & val.value) == val.value; }
```
unrelated note: `properties.deviceType` is for the physical device and it's a single bit (not a bitflag, just enum)

Reply:
okay